src/resources.py

```diff
@@ -887,23 +960,34 @@ def _init_ui(self):
 
         # --------- table_workspace ------------
 
-        table_workspace, workspace_layout = add_frame(self, height=72, name="Page_table_workspace",
-                                                      l_m=40, r_m=40, t_m=21, b_m=21, space=32, layout=HORIZONTAL)
+        table_workspace, workspace_layout = add_frame(self, height=102, name="Page_table_workspace", layout=VERTICAL)
         window_layout.addWidget(table_workspace)
 
-        self.search_bar = add_input(table_workspace, height=30, name="Page_table_workspace_search",
+        feature, feature_layout = add_frame(table_workspace, height=72, l_m=40, r_m=40, t_m=26, b_m=11, space=32, layout=HORIZONTAL)
+        workspace_layout.addWidget(feature)
+
+        hint_frame, hint_layout = add_frame(table_workspace, height=30, l_m=40, r_m=40, b_m=5, layout=HORIZONTAL)
+        workspace_layout.addWidget(hint_frame)
+
+        self.search_bar = add_input(feature, height=35, name="Page_table_workspace_search",
                                     hint="Search a machine... (Haven't implemented yet)")
-        workspace_layout.addWidget(self.search_bar)
+        feature_layout.addWidget(self.search_bar)
+
+        self.refresh_button = add_button(feature, "REFRESH", name="Page_table_workspace_button")
```

Comment:
I tried it today, it doesn't work properly. It is not anything about `PyQt` In the normal cases, python is passed by value. when we assigned return value from `add_button` to `button`, it is not actually assigned to ` self.remove_button`. Therefore, when we called it somewhere else, it would raise `AttributeError` and tell `NoneType` object has no attribute xxx.